core/app/models/spree/state.rb

```diff
@@ -6,7 +6,7 @@ class State < Spree::Base
     validates :country, :name, presence: true
 
     def self.find_all_by_name_or_abbr(name_or_abbr)
-      where('name = ? OR abbr = ?', name_or_abbr, name_or_abbr)
+      where('lower(name) = lower(:name_or_abbr) OR lower(abbr) = lower(:name_or_abbr)', name_or_abbr: name_or_abbr)
```

Comment:
Rather than using `lower` using arel matches would be preferred. This will use `ILIKE` in postgres, and `LIKE` in MySQL.

These types of comparisons can make better use of indexes than relying upon the SQL function return value.

Example:

```Ruby
where(
  arel_table[:name].matches(name_or_abbr).or(
    arel_table[:abbr].matches(name_or_abbr)
  )
)
```